From dcca34a74b1b8e208c5292d4e731bd097e118f08 Mon Sep 17 00:00:00 2001
From: Gurchetan Singh <gurchetansingh@chromium.org>
Date: Fri, 15 Mar 2019 10:08:45 -0700
Subject: [PATCH 2/6] configure.ac: depend on libnativewindow when appropriate

libnativewindow is only available on O or greater, and it's
required for some features.

Change-Id: Ifd4c093049dbad726f0835b9338396cd7502d86c
---
 configure.ac | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index e5c55ef9e0f0..fc0d56b6ad50 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1573,6 +1573,12 @@ if test "x$enable_libglvnd" = xyes -a "x$EGL_LIB_SUFFIX" != x; then
     AC_MSG_ERROR([EGL lib suffix can't be used with libglvnd])
 fi
 
+AC_ARG_WITH([platform-sdk-version],
+  [AS_HELP_STRING([--with-platform-sdk-version@<:@=NAME@:>@],
+    [specify Android platform SDK version @<:@default=none@:>@])],
+  [platform_sdk_version=$withval],
+  [platform_sdk_version=""])
+
 dnl
 dnl Mangled Mesa support
 dnl
@@ -1907,7 +1913,12 @@ for plat in $platforms; do
         ;;
 
     android)
-        PKG_CHECK_MODULES([ANDROID], [cutils hardware sync])
+        if test -n "$platform_sdk_version" && test "$platform_sdk_version" -ge 26; then
+            PKG_CHECK_MODULES([ANDROID], [cutils hardware nativewindow sync])
+        else
+            PKG_CHECK_MODULES([ANDROID], [cutils hardware sync])
+        fi
+
         if test -n "$with_gallium_drivers"; then
             PKG_CHECK_MODULES([BACKTRACE], [backtrace])
         fi
-- 
2.20.1

